upgrade/upgrade-2.1.2.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+// In the latest version PrestaShop 1.7.8 the override for this module in classic theme was removed
+// because the module is now self-sufficient, so we remove it from the theme to make sure the correct
+// internal template is used (we only clean the theme from core classic theme, and only if it was not
+// updated). Since this version of the module is only compatible starting PS 1.7.8 this clean can always
+// be performed regardless of the current PrestaShop version.
+// We check that the file matches the latest MD5 from this PR https://github.com/PrestaShop/PrestaShop/pull/22032
+// when it was removed
+
+if (defined('_PS_ROOT_DIR_')) {
+    $coreThemeFileChecksum = 'a37c4a845628588260b776d747905b0a';
```

Comment:
Sorry but, I'm afraid it can't work like that.
By doing this, you accept only the upgrade from 1.7.7.x:

1.7.6.x: df519b7b09781d5d15c16358c1145fd4  ps_searchbar.tpl
1.7.5.x: ae5932a8e0ec09ab9b11e68398e636e3  ps_searchbar.tpl

And it changes each time one character (year number in the license for example), has been updated :p

Reply:
I know but the core files are supposed to be updated before the modules right?

And this new version of ps_searchbar is only compatible with PS >1.7.8, so when the module is upgraded it should already be PS 1.7.8 minimum

Reply:
@jolelievre Could you add a comment for explaining in code the expected behavior ?